drivers/sensor/adxl367/adxl367.c

```diff
@@ -931,6 +938,14 @@ static int adxl367_probe(const struct device *dev)
 		return ret;
 	}
 
+#ifdef CONFIG_ADXL367_TRIGGER
+	ret = adxl367_init_interrupt(dev);
+	if (ret) {
```

Comment:
Here and all over this PR, this isn't compliant with the Zephyr coding guidelines, please use an explicit `ret != 0`

See https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#:~:text=The%20controlling%20expression%20of%20an%20if%20statement%20and%20the%20controlling%20expression%20of%20an%20iteration%2Dstatement%20shall%20have%20essentially%20Boolean%20type

And https://gitlab.com/MISRA/MISRA-C/MISRA-C-2012/Example-Suite/-/blob/master/R_14_04.c#L44